src/engine.rs

```diff
@@ -468,6 +468,7 @@ where
     /// Like `cut_logs` but the range could be very large. Return the deleted count.
     /// Generally, `from` can be passed in `0`.
     pub fn compact_to(&self, region_id: u64, index: u64) -> u64 {
+        // FIXME: potential race?
```

Comment:
I think purging can handle with stale indexes, but they may lead to inaccurate metrics.